### DISCUSSION TITLE
[Feature] Optionally load full original image

### DISCUSSION THREAD

**Discussion by dvdkon** [immich-app/immich#3336] [Feature Request]
### The feature

Currently, the web app (and I presume the mobile app) loads downsized and converted versions of images. I would like to be able to view the full, original versions of these images inside the existing apps when possible (i.e. the browser/mobile app can decode the image format natively).

I propose adding either a toggle to the user's settings or a toggle button directly to the asset viewer. Ideally the current resized asset would load first, with the full image replacing it once loaded.

My usecase is browsing my library on the local network, where loading the original is near-instant, making the use of smaller images pointless.

I can implement this on the web client, but I'd like to have consensus on how it should work first.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
I believe the mobile app already has a local setting for this. I'm sure it would be worth adding one on the web client as well.

**Reply by dvdkon** [reply]
You're right, I must have missed it. Do you think it should be a per-user toggle, per-browser toggle (i.e. localStorage) or a per-session one directly accessible from the viewer? I think per-browser would be best (my desktop will always be on the same network), but it might confuse users that there are multiple types of settings if it was in the settings menu.

**Reply by bo0tzz** [reply]
I think per-browser makes the most sense conceptually, but we don't have anything in place for settings like that yet whereas we do for per-user. @alextran1502 @jrasm91 thoughts?

**Reply by zackpollard** [reply]
Personally I'm not sure showing the actual original file makes sense in-browser as there are plenty of formats that won't be supported. In my opinion the benefit of a tool like immich is that we have full compatibility with all clients for all files. I'm dubious about adding support for something like this, as longer term I think the right direction to go is to have converted files that are displayed in all cases for maximum compatibility. We could I guess have the option for the browser to try and load the original, with the webp as a backup, but if we do the conversion stuff right (like Google photos does) you actually shouldn't notice the difference in most cases. If you want the actual original, it's there and you can download it

**Reply by dvdkon** [reply]
I think it would be wasteful to always convert for compatibility, when plenty of files (anything but HEIF, really) are directly compatible. Besides, a smart implementation could always convert when necessary and show the original when not. But my real need is to show an image that has no quality loss compared to the original, so I suppose I'm fine either way.

**Reply by jrasm91** [reply]
I would maybe support the idea of loading the original in the asset viewer in some cases, but I don't think it is worthwhile to have separate settings, code, query params, etc. for loading originals for thumbnails. It will probably add a decent amount of complexity to the code and I'm not sure it will have perceivable benefit.

**Reply by mertalev** [reply]
I think @uhthomas mentioned at some point that we should ideally have different thumbnail/image sizes for different clients. The logic for handling that could also handle selecting the original as an option.

**Reply by uhthomas** [reply]
https://github.com/immich-app/immich/issues/2749

**Comment by dvdkon** [comment]
I updated my install and it turns out @lmichelin had recently implemented this in bf3b38a7f29f33d8aef207e84055caa8fc7237b3. Thanks!
